diff --git a/win32/resource.rb b/win32/resource.rb
index 6468233..5cad6ed 100755
--- a/win32/resource.rb
+++ b/win32/resource.rb
@@ -52,13 +52,16 @@ def icons.find(path)
   [$rubyw_name,  CONFIG["EXEEXT"], 'VFT_APP', 'GUI', rubyw_icon || ruby_icon],
   [$so_name,     '.dll',           'VFT_DLL', 'DLL', dll_icons.join],
 ].each do |base, ext, type, desc, icon|
+  manifest = (type=='VFT_APP') ? "1" : "2"
   open(base + '.rc', "w") { |f|
     f.binmode if /mingw/ =~ RUBY_PLATFORM
 
     f.print <<EOF
 #include <windows.h>
 #include <winver.h>
+#include <winuser.h>
 
+#{manifest} RT_MANIFEST "ruby.bin.manifest"
 #{icon || ''}
 VS_VERSION_INFO VERSIONINFO
  FILEVERSION    #{nversion}
@@ -93,3 +96,60 @@ def icons.find(path)
   }
 end
 
+manifest_version = RUBY_VERSION + ".0"
+manifest_arch = (CONFIG["target_cpu"] == "x64") ? 'amd64' :
+  (CONFIG["target_cpu"] =~ /i[3456]86/) ? 'x86' :
+  'error'
+
+open('ruby.bin.manifest', "w") { |f|
+    f.binmode if /mingw/ =~ RUBY_PLATFORM
+
+    f.print <<EOF_MANI_BIN
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+
+  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+      <application>
+        <!-- Windows Vista -->
+          <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}"/>
+        <!-- Windows 7 -->
+          <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}"/>
+        <!-- Windows 8 -->
+          <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}"/>
+        <!-- Windows 8.1 -->
+          <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}"/>
+        <!-- Windows 10 -->
+          <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}"/>
+      </application>
+  </compatibility>
+
+  <assemblyIdentity type="win32"
+                    name="RubyInstaller.MRI.Ruby"
+                    version="#{manifest_version}"
+                    processorArchitecture="#{manifest_arch}"
+  />
+
+  <dependency>
+    <dependentAssembly>
+      <assemblyIdentity type="win32"
+                        name="RubyInstaller.MRI.RubyAssembly"
+                        version="#{manifest_version}"
+                        language="*"
+                        processorArchitecture="#{manifest_arch}" />
+    </dependentAssembly>
+  </dependency>
+</assembly>
+EOF_MANI_BIN
+}
+
+open('RubyInstaller.MRI.RubyAssembly.manifest', "w") { |f|
+    f.binmode if /mingw/ =~ RUBY_PLATFORM
+
+    f.print <<EOF_MANI_ASSY
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+    <assemblyIdentity type="win32" name="RubyInstaller.MRI.RubyAssembly" version="#{manifest_version}" processorArchitecture="#{manifest_arch}" />
+
+<!-- PLACE_HOLDER -->
+</assembly>
+EOF_MANI_ASSY
+}
