diff -ru a/src/x86/ffi.c b/src/x86/ffi.c
--- a/src/x86/ffi.c	2010-01-01 01:32:40.000000000 +0900
+++ b/src/x86/ffi.c	2011-05-08 22:44:45.985753600 +0900
@@ -262,7 +262,7 @@
 #endif /* X86_WIN32 */
 #ifdef X86_WIN64
 extern int
-ffi_call_win64(void (*)(char *, extended_cif *), extended_cif *,
+_ffi_call_win64(void (*)(char *, extended_cif *), extended_cif *,
                unsigned, unsigned, unsigned *, void (*fn)(void));
 #endif
 
@@ -317,7 +317,7 @@
               avalue[i] = local;
             }
         }
-        ffi_call_win64(ffi_prep_args, &ecif, cif->bytes,
+        _ffi_call_win64(ffi_prep_args, &ecif, cif->bytes,
                        cif->flags, ecif.rvalue, fn);
       }
       break;
@@ -355,14 +355,14 @@
      __attribute__ ((regparm(1)));
 #endif
 #ifdef X86_WIN64
-void FFI_HIDDEN ffi_closure_win64 (ffi_closure *);
+void FFI_HIDDEN _ffi_closure_win64 (ffi_closure *);
 #endif
 
 /* This function is jumped to by the trampoline */
 
 #ifdef X86_WIN64
 void * FFI_HIDDEN
-ffi_closure_win64_inner (ffi_closure *closure, void *args) {
+_ffi_closure_win64_inner (ffi_closure *closure, void *args) {
   ffi_cif       *cif;
   void         **arg_area;
   void          *result;
@@ -545,7 +545,7 @@
     {
       int mask = FLAG(0)|FLAG(1)|FLAG(2)|FLAG(3);
       FFI_INIT_TRAMPOLINE_WIN64 (&closure->tramp[0],
-                                 &ffi_closure_win64,
+                                 &_ffi_closure_win64,
                                  codeloc, mask);
       /* make sure we can execute here */
     }
