From 67d50ccab7e86491a1aa2347e00530142c083c15 Mon Sep 17 00:00:00 2001
From: Alexey Borzenkov <snaury@gmail.com>
Date: Sun, 16 Aug 2009 01:50:09 +0400
Subject: [PATCH 2/2] On msys perl symlink does not work

---
 util/mklink.pl |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/util/mklink.pl b/util/mklink.pl
index eacc327..2f3ef7b 100644
--- a/util/mklink.pl
+++ b/util/mklink.pl
@@ -58,7 +58,7 @@ foreach $dirname (@from_path) {
 my $to = join('/', @to_path);
 
 my $file;
-$symlink_exists=eval {symlink("",""); 1};
+$symlink_exists=$^O != "msys" && eval {symlink("",""); 1};
 foreach $file (@files) {
     my $err = "";
     if ($symlink_exists) {
-- 
1.6.4

