diff --git a/win32/resource.rb b/win32/resource.rb
index 786edb0..0970e15 100755
--- a/win32/resource.rb
+++ b/win32/resource.rb
@@ -52,6 +52,7 @@ end
   [$rubyw_name,  CONFIG["EXEEXT"], 'VFT_APP', 'GUI', rubyw_icon || ruby_icon],
   [$so_name,     '.dll',           'VFT_DLL', 'DLL', dll_icons.join],
 ].each do |base, ext, type, desc, icon|
+  manifest = (type=='VFT_APP') ? "1" : "2"
   open(base + '.rc', "w") { |f|
     f.binmode if /mingw/ =~ RUBY_PLATFORM
 
@@ -59,8 +60,10 @@ end
 #ifndef __BORLANDC__
 #include <windows.h>
 #include <winver.h>
+#include <winuser.h>
 #endif
 
+#{manifest} RT_MANIFEST "ruby.bin.manifest"
 #{icon || ''}
 VS_VERSION_INFO VERSIONINFO
  FILEVERSION    #{nversion}
@@ -95,3 +98,45 @@ EOF
   }
 end
 
+manifest_version = RUBY_VERSION + ".0"
+manifest_arch = (CONFIG["target_cpu"] == "x64") ? 'amd64' :
+  (CONFIG["target_cpu"] =~ /i[3456]86/) ? 'x86' :
+  'error'
+
+open('ruby.bin.manifest', "w") { |f|
+    f.binmode if /mingw/ =~ RUBY_PLATFORM
+
+    f.print <<EOF_MANI_BIN
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+
+  <assemblyIdentity type="win32"
+                    name="RubyInstaller.MRI.Ruby"
+                    version="#{manifest_version}"
+                    processorArchitecture="#{manifest_arch}"
+  />
+
+  <dependency>
+    <dependentAssembly>
+      <assemblyIdentity type="win32"
+                        name="RubyInstaller.MRI.RubyAssembly"
+                        version="#{manifest_version}"
+                        language="*"
+                        processorArchitecture="#{manifest_arch}" />
+    </dependentAssembly>
+  </dependency>
+</assembly>
+EOF_MANI_BIN
+}
+
+open('RubyInstaller.MRI.RubyAssembly.manifest', "w") { |f|
+    f.binmode if /mingw/ =~ RUBY_PLATFORM
+
+    f.print <<EOF_MANI_ASSY
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+    <assemblyIdentity type="win32" name="RubyInstaller.MRI.RubyAssembly" version="#{manifest_version}" processorArchitecture="#{manifest_arch}" />
+
+<!-- PLACE_HOLDER -->
+</assembly>
+EOF_MANI_ASSY
+}
